cassandra-analytics-release/prepare_release.sh

```diff
@@ -151,35 +151,38 @@ then
-    echo "The release candidate for ${release} is already staged at https://dist.apache.org/repos/dist/dev/cassandra/cassandra-analytics/${release}"
-    exit 1
-fi
-
-if curl --output /dev/null --silent --head --fail "https://archive.apache.org/dist/cassandra/cassandra-analytics/${release}" ; then
-    echo "A published release for ${release} is already public at https://archive.apache.org/dist/cassandra/cassandra-analytics/${release}"
-    exit 1
-fi
-
-if curl --output /dev/null --silent --head --fail "https://github.com/apache/cassandra-analytics/tree/${release}-tentative" ; then
-    echo "The release candidate tag for ${release}-tentative is already at https://github.com/apache/cassandra-analytics/tree/${release}-tentative"
-    exit 1
-fi
-
-if curl --output /dev/null --silent --head --fail "https://github.com/apache/cassandra-analytics/tree/cassandra-analytics-${release}" ; then
-    echo "The published release tag for ${release} is already at https://github.com/apache/cassandra-analytics/tree/cassandra-analytics-${release}"
-    exit 1
-fi
-
-if git tag -l | grep -q "${release}-tentative"; then
-    echo "Local git tag for ${release}-tentative already exists"
-    exit 1
+if [ $fake_mode -ne 1 ]
+then
+  gradle_properties_version="$(grep ^version= gradle.properties)"
+  if [ "${release}" != "${gradle_properties_version#version=}" ] ; then
+      echo "The release requested ${release} does not match gradle.properties's version ${gradle_properties_version}"
+      exit 1
+  fi
+
+  if curl --output /dev/null --silent --head --fail "https://dist.apache.org/repos/dist/dev/cassandra/cassandra-analytics/${release}" ; then
+      echo "The release candidate for ${release} is already staged at https://dist.apache.org/repos/dist/dev/cassandra/cassandra-analytics/${release}"
+      exit 1
+  fi
+
+  if curl --output /dev/null --silent --head --fail "https://archive.apache.org/dist/cassandra/cassandra-analytics/${release}" ; then
+      echo "A published release for ${release} is already public at https://archive.apache.org/dist/cassandra/cassandra-analytics/${release}"
+      exit 1
+  fi
+
+  if curl --output /dev/null --silent --head --fail "https://github.com/apache/cassandra-analytics/tree/${release}-tentative" ; then
+      echo "The release candidate tag for ${release}-tentative is already at https://github.com/apache/cassandra-analytics/tree/${release}-tentative"
+      exit 1
+  fi
+
+  if curl --output /dev/null --silent --head --fail "https://github.com/apache/cassandra-analytics/tree/cassandra-analytics-${release}" ; then
+      echo "The published release tag for ${release} is already at https://github.com/apache/cassandra-analytics/tree/cassandra-analytics-${release}"
+      exit 1
+  fi
+
+  if git tag -l | grep -q "${release}-tentative"; then
+      echo "Local git tag for ${release}-tentative already exists"
+      exit 1
+  fi
```

Comment:
Not sure why this block is prevented from running in "fake" mode (maybe rename it to "dry-run" mode as I think that's a more typical name for this kind of thing)? Everything here is just checking parameters, which seems like something you'd want to do in dry-run mode.